drivers/hmis/spec/jobs/merge_clients_job_spec.rb

```diff
@@ -9,279 +9,594 @@
+        let!(:external_id_client_1) { create :mci_external_id, source: client1, remote_credential: mci_cred, value: mci_id_value }
+        let!(:external_id_client_2) { create :mci_external_id, source: client2, remote_credential: mci_cred, value: mci_id_value }
+
+        it 'merges and deduplicates external ids' do
+          Hmis::MergeClientsJob.perform_now(client_ids: client_ids, actor_id: actor.id)
+          expect(client1.ac_hmis_mci_ids.pluck(:value)).to contain_exactly(mci_id_value)
+        end
+      end
+
+      context 'with unique mci_ids' do
+        let!(:record1) { create :mci_external_id, source: client1, remote_credential: mci_cred }
+        let!(:record2) { create :mci_external_id, source: client2, remote_credential: mci_cred }
+
+        it 'merges external ids' do
+          Hmis::MergeClientsJob.perform_now(client_ids: client_ids, actor_id: actor.id)
+          expect(client1.ac_hmis_mci_ids.pluck(:value)).to contain_exactly(record1.value, record2.value)
+        end
+
+        it_behaves_like 'merge that saves mappings', 'mci_ids', 'source_id', 'id'
+      end
+
+      context 'where merged client has mci_unique_id' do
+        let!(:record2) { create :mci_unique_id_external_id, source: client2, remote_credential: mci_unique_id_cred }
+
+        it 'retains mci_unique_id' do
+          Hmis::MergeClientsJob.perform_now(client_ids: client_ids, actor_id: actor.id)
+          expect(client1.ac_hmis_mci_unique_id&.value).to eq(record2.value)
+        end
+
+        it_behaves_like 'merge that saves mappings', 'mci_unique_ids', 'source_id', 'id'
+      end
+
+      context 'where retained client has mci_unique_id' do
+        let!(:record2) { create :mci_unique_id_external_id, source: client1, remote_credential: mci_unique_id_cred }
+
+        it 'retains mci_unique_id' do
+          Hmis::MergeClientsJob.perform_now(client_ids: client_ids, actor_id: actor.id)
+          expect(client1.ac_hmis_mci_unique_id&.value).to eq(record2.value)
+        end
+      end
+
+      context 'where clients have different mci_unique_ids' do
+        let!(:record1) { create :mci_unique_id_external_id, source: client1, remote_credential: mci_unique_id_cred }
+        let!(:record2) { create :mci_unique_id_external_id, source: client2, remote_credential: mci_unique_id_cred }
+
+        it 'retains one mci_unique_id' do
+          expect do
+            Hmis::MergeClientsJob.perform_now(client_ids: client_ids, actor_id: actor.id)
+          end.to change(HmisExternalApis::ExternalId.mci_unique_ids, :count).by(-1)
+          expect(client1.ac_hmis_mci_unique_id&.value).to eq(record1.value)
+        end
+
+        # TODO- discuss, do we need to add this expectation? I think no, because in this case the discarded unique ID is not updated.
+        # but I think it's fully deleted (not soft deleted) so this might be something to flag for unmerge
+        # it_behaves_like 'merge that saves mappings', 'mci_unique_ids', 'source_id', 'id'
```

Comment:
Do we need to make updates to mci unique ids so that they are fully recoverable when unmerging?